sdk/bootstrap/main.go

```diff
@@ -132,47 +197,62 @@ func waitForResolve(resolveHosts []string, resolveTimeout time.Duration) {
 	} else {
 		timer = time.NewTimer(resolveTimeout)
 	}
+
 	for _, host := range resolveHosts {
-		log.Printf("Waiting for '%s' to resolve...", host)
-		for {
-			result, err := net.LookupHost(host)
-
-			// Check result, exit loop if suceeded:
-			if err != nil {
-				if verbose {
-					log.Printf("Lookup failed: %s", err)
-				}
-			} else if len(result) == 0 {
-				if verbose {
-					log.Printf("No results for host '%s'", host)
-				}
-			} else {
-				log.Printf("Resolved '%s' => %s", host, result)
-				break
-			}
+		resolveHost(host, timer)
+	}
+
+	if verbose {
+		log.Printf("Hosts resolved, continuing bootstrap.")
+	}
+
+	// Clean up:
+	if !timer.Stop() {
+		<-timer.C
+	}
+}
+
+func verifySelfResolution(podIP string, resolveTimeout time.Duration) {
+	taskHost, err := parseTaskHost()
+	if err != nil {
+		printEnv()
+		log.Printf("Missing required envvars to build task DNS address. " +
+			"Ensure that TASK_NAME and FRAMEWORK_HOST are both set or " +
+			"disable self resolution with --self-resolve=false")
+	}
+	log.Printf("Verifying that %s resolves to %s", taskHost, podIP)
 
-			// Check timeout:
-			if timer != nil {
-				select {
-				case _, ok := <-timer.C:
-					if ok {
-						log.Fatalf("Time ran out while resolving '%s'. "+
-							"Customize timeout with -resolve-timeout, or use -verbose to see attempts.", host)
-					} else {
-						log.Fatalf("Internal error: Channel closed")
-					}
-				default:
-					// do nothing
-				}
+	var timer *time.Timer
+	if resolveTimeout == 0 {
+		timer = nil
+	} else {
+		timer = time.NewTimer(resolveTimeout)
+	}
+
+	for {
+		resolvedIPs := resolveHost(taskHost, timer)
```

Comment:
Why is this treated as such as special case? Is this not the same as calling `waitForResolve()` with a list containing the `taskHost` (with the modification that we need add return values from `waitForResolve()`?

Reply:
I want the custom log lines around resolving to something specific.